sympy/polys/appellseqs.py

```diff
@@ -0,0 +1,156 @@
+"""Efficient functions for generating Appell sequences."""
+from sympy.core.symbol import Dummy
+from sympy.polys.densearith import dup_mul_ground, dup_add_ground
+from sympy.polys.densetools import dup_eval, dup_integrate
+from sympy.polys.domains import ZZ, QQ
+from sympy.polys.polyclasses import DMP
+from sympy.polys.polytools import Poly, PurePoly
+from sympy.utilities import public
+
+def dup_appell(n, seq, v, f, K):
+    """Common low-level implementation of Appell sequences."""
+    for i in range(2, n+1):
+        q = dup_integrate(dup_mul_ground(seq[-1], i, K), 1, K)
+        seq.append(dup_add_ground(q, f(dup_eval(q, v, K), i), K))
+    return seq[n]
+
+@public
+def appell_poly(n, seq, v, f, K, x=None, polys=False):
```

Comment:
This function seems a bit complicated for a public function. At least it would need more explanation or a doctest or something.

I checked both of the citations and neither of them gives a clear explanation as to where the parameters `f` and `v` come from here. Presumably this characterisation of Appell sequences and this algorithm is described somewhere but it doesn't seem to be in those citations.

Also the description of seq is not completely clear e.g. should it be `Expr` or domain coefficients? Passing the domain `K` suggests that we're working in the poly domains where a list of domain elements is the DUP representation of a polynomial but I think the intention here is that a list of lists of Expr should be passed for this parameter.

Reply:
Right, perhaps I tried too hard to shoehorn everything into the same framework, eh?

Reply:
I thought that the structure of the code was fine but that the `appell_poly` function didn't seem useful as a public function because it seemed unlikely that anyone would understand how to use it.

The new code does seem clearer to me though.